### PR TITLE
Introduce scoped by component configuration

### PR DESCRIPTION
**Pull request by metalmatze**
We want to make it easier to configure kube-thanos with common variables like name, namespace, image and replicas.

/cc @s-urbaniak @kakkoyun @squat @brancz

### CODE REVIEW

**Comment by s-urbaniak** [comment]
looks good from my side, thanks! :+1:

File: README.md [hunk 1]
```diff
@@ -81,21 +81,17 @@ local kt =
   // (import 'kube-thanos/kube-thanos-servicemonitors.libsonnet') +
   {
     thanos+:: {
-      variables+:: {
-        image: 'quay.io/thanos/thanos:v0.8.0',
-        objectStorageConfig+: {
-          name: 'thanos-objectstorage',
-          key: 'thanos.yaml',
-        },
+      image:: 'quay.io/thanos/thanos:v0.8.0',
```
[the remark this answers is not in the document]
Comment (brancz) [reply README.md]: generally curious, why don't we enforce this to be set by the user, by having the default `error "must provide image"` here? Seems like this kind of thing shouldn't need to be updated in this repo for every release.

[the remark this answers is not in the document]
Comment (metalmatze) [reply README.md]: This is now enforced in the library itself:
https://github.com/thanos-io/kube-thanos/pull/55/files#diff-2a5f73045379318b1c47c770210b5dffR6

But this is the entry example and thus I want to provide people with something that works. The actual version doesn't matter at all.

[the remark this answers is not in the document]
Comment (metalmatze) [reply README.md]: Updated the example.jsonnet and all.jsonnet with a comment explaining that any image can be set here and it's just an example.

**Pull request merged**